Autopilot/boardfiles/Src/freertos.c

```diff
@@ -136,40 +152,115 @@ void MX_FREERTOS_Init(void) {
   /* USER CODE END RTOS_QUEUES */
 
   /* Create the thread(s) */
-  /* definition and creation of defaultTask */
-  osThreadDef(defaultTask, StartDefaultTask, osPriorityNormal, 0, 128);
-  defaultTaskHandle = osThreadCreate(osThread(defaultTask), NULL);
+  /* definition and creation of attitudeManager */
+  osThreadDef(attitudeManager, attitudeManagerExecute, osPriorityNormal, 0, 128);
+  attitudeManagerHandle = osThreadCreate(osThread(attitudeManager), NULL);
 
   /* definition and creation of Interchip */
   osThreadDef(Interchip, Interchip_Run, osPriorityNormal, 0, 128);
   InterchipHandle = osThreadCreate(osThread(Interchip), NULL);
 
-  /* definition and creation of Attitude */
-  osThreadDef(Attitude, Attitude_Run, osPriorityNormal, 0, 128);
-  AttitudeHandle = osThreadCreate(osThread(Attitude), NULL);
+  /* definition and creation of pathManager */
+  osThreadDef(pathManager, pathManagerExecute, osPriorityBelowNormal, 0, 128);
+  pathManagerHandle = osThreadCreate(osThread(pathManager), NULL);
+
+  /* definition and creation of telemetryRun */
+  osThreadDef(telemetryRun, StartTelemetryRun, osPriorityBelowNormal, 0, 128);
```

Comment:
good stuff. Just a question, "osPriorityBelowNorma" what does that mean and why did you chose it ?

Reply:
tbh this was a pretty arbitrary decision, there are different priority levels for the threads, where osPriorityRealtime is the highest and osPriorityBelowNormal is the lowest. I guess it depends on how important sensorfusion is compared to attitudemanager, telem, and pathmanager which would also influence the priorities of those ones. Would you have any suggestions for the priority?

Reply:
Ah I see. If we do things right, threads should never step on each other's toes. In fact, we should make sure that each thread has finished a single execution before FREERTOS moves on to the next one. That will likely be done via a mutex. But we should talk about that later. ATM, having them all normal is fine